### ISSUE TITLE
TypeError: Cannot read property 'js' of undefined

### ISSUE THREAD

**Issue by rouwe111** [ztoben/assets-webpack-plugin#125]
**Describe the bug**

It throws an error when building.

```
/Users/user/Web/node_modules/assets-webpack-plugin/dist/index.js:83
            var type = _typeof(typeMap[typeName]);
                                      ^

TypeError: Cannot read property 'js' of undefined
    at /Users/user/Web/node_modules/assets-webpack-plugin/dist/index.js:83:39
    at Array.reduce (<anonymous>)
    at /Users/user/Web/node_modules/assets-webpack-plugin/dist/index.js:74:30
    at Array.reduce (<anonymous>)
    at afterEmit (/Users/user/Web/node_modules/assets-webpack-plugin/dist/index.js:67:27)
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/Users/user/Web/node_modules/tapable/lib/HookCodeFactory.js:24:12), <anonymous>:12:1)
    at AsyncSeriesHook.lazyCompileHook [as _callAsync] (/Users/user/Web/node_modules/tapable/lib/Hook.js:35:21)
    at asyncLib.forEach.err (/Users/user/Web/node_modules/webpack/lib/Compiler.js:355:27)
    at done (/Users/user/Web/node_modules/neo-async/async.js:2854:11)
    at /Users/user/Web/node_modules/neo-async/async.js:2805:7
    at /Users/user/Web/node_modules/graceful-fs/graceful-fs.js:43:10
```

**To Reproduce**
1. Run the Webpack build
2. See error

**Expected behavior**
I expect to create an assets.json file with all js and css files (since this is the default for the fileTypes option).

**Webpack Config**
The webpack configuration you are using when seeing the bug.

```
const path = require('path');
const webpack = require('webpack');
const merge = require('webpack-merge');
const CompressionPlugin = require('compression-webpack-plugin');
const AssetsPlugin = require('assets-webpack-plugin');
const MiniCssExtractPlugin = require('mini-css-extract-plugin');
const baseConfig = require('./webpack.config.base');
const options = require('./webpack.options');

const productionConfig = merge(baseConfig, {
    entry: [
        'babel-polyfill',
        path.join(process.cwd(), 'src/Client/index.js')
    ],

    optimization: {
        minimize: true,
        nodeEnv: 'production',
        sideEffects: true,
        concatenateModules: true,
        splitChunks: {
            name: false,
            cacheGroups: {
                vendors: {
                    test: /[\\/]node_modules[\\/]/,
                    name: 'vendor',
                    chunks: 'all'
                }
            }
        },
        runtimeChunk: {
            name: 'manifest'
        }
    },

    module: {
        rules: [
            {
                test: /\.js$/,
                loader: 'babel-loader',
                exclude: /node_modules/,
                include: options.clientInclude
            },
            {
                test: /(\.css|\.scss)$/,
                use: [
                    MiniCssExtractPlugin.loader,
                    {
                        loader: 'css-loader',
                        options: {
                            minimize: true,
                            sourceMap: false
                        }
                    },
                    {
                        loader: 'postcss-loader'
                    },
                    {
                        loader: 'sass-loader',
                        options: {
                            outputStyle: 'expanded',
                            sourceMap: false,
                            sourceMapContents: false
                        }
                    }
                ]
            },
            {
                test: /\.(png|jpg|gif|svg|jpe?g)$/,
                loader: 'file-loader',
                exclude: /node_modules/,
                options: {
                    name: 'images/[name].[ext]?[hash]'
                }
            },
            {
                test: /\.(ttf|eot|woff|woff2)$/,
                loader: 'file-loader',
                exclude: /node_modules/,
                options: {
                    name: 'fonts/[name].[ext]?[hash]'
                }
            }
        ]
    },

    plugins: [
        new webpack.DefinePlugin({
            '__CLIENT__': true,
            '__SERVER__': false
        }),

        new webpack.optimize.AggressiveMergingPlugin(),

        new webpack.HashedModuleIdsPlugin({
            hashFunction: 'sha256',
            hashDigest: 'hex',
            hashDigestLength: 20
        }),

        new MiniCssExtractPlugin({
            filename: 'css/[name].[contenthash].css'
        }),

        new CompressionPlugin({
            asset: '[path].gz[query]',
            algorithm: 'gzip',
            test: new RegExp('\\.(js|css)$'),
            threshold: 10240,
            minRatio: 0.8
        }),

        new AssetsPlugin({
            path: options.buildPath,
            filename: 'assets.json',
            includeManifest: 'manifest',
            includeAllFileTypes: false
        })
    ],

    performance: {
        assetFilter: assetFilename => !(/(\.map$)|(^(main\.|favicon\.))/).test(assetFilename)
    }
});

module.exports = productionConfig;
```

**Desktop (please complete the following information):**
 - OS: macOS High Sierra 10.13.5
 - Node version: 8.11.2
 - Plugin version: 3.9.4 and 3.9.5

**Comment by ztoben** [comment]
Thanks for the report, I'll take a look when I get a chance.

**Comment by rouwe111** [comment]
Thanks! I'll stick to 3.9.3 in the meantime, which works perfectly 👍

**Comment by ztoben** [comment]
Does `3.9.5` work if you don't set `includeAllFileTypes`?

**Comment by ztoben** [comment]
@rouwe111, You mind trying `3.9.5-alpha.2` and let me know how that goes? Thanks!

**Comment by rouwe111** [comment]
@ztoben `3.9.5` without `includeAllFileTypes` works fine. `3.9.5-alpha.2` including `includeAllFileTypes` also works fine.

**Comment by ztoben** [comment]
Okay, I'll push out a new version with the changes from the alpha then. I was accidentally returning an undefined value in the assets reducer. I'll close this when the new version with the fix is out, `v3.9.6`.

**Comment by rouwe111** [comment]
Thank you @ztoben !
